bcda/auth/plugin/alpha.go

```diff
@@ -248,18 +247,65 @@ func revokeAccessTokenByID(tokenID uuid.UUID) error {
 	return db.Error
 }
 
-func (p *AlphaAuthPlugin) ValidateAccessToken(token string) error {
-	return errors.New("not yet implemented")
+func (p *AlphaAuthPlugin) ValidateAccessToken(tokenString string) error {
+	t, err := p.DecodeAccessToken(tokenString)
+	if err != nil {
+		return err
+	}
+
+	c := t.Claims.(*AllClaims)
+
+	err = checkRequiredClaims(c)
+	if err != nil {
+		return err
+	}
+
+	err = c.Valid()
+	if err != nil {
+		return err
+	}
+
+	_, err = getACOFromDB(c.ACO)
+	if err != nil {
+		return err
+	}
+
+	b := isBlacklisted(t)
+	if b {
+		return fmt.Errorf("blacklisted Token with ID: %v was rejected", c.ID)
+	}
+
+	return nil
+}
+
+func checkRequiredClaims(claims *AllClaims) error {
+	if claims.ExpiresAt == 0 ||
+		claims.IssuedAt == 0 ||
+		claims.Subject == "" ||
+		claims.ACO == "" ||
+		claims.ID == "" {
+		return fmt.Errorf("missing one or more required claims")
+	}
+	return nil
+}
+
+func isBlacklisted(token jwt.Token) bool {
+	c := token.Claims.(*AllClaims)
+
+	db := database.GetGORMDbConnection()
+	defer db.Close()
+
+	return !db.Find(&token, "UUID = ? AND active = ?", c.ID, false).RecordNotFound()
```

Comment:
This function will return false when the token id is not found in the db. That's not a Good Thing.

Reply:
Then someone would be allowed to use the system if they submit a token with a non-existent database id. That's not good in any circumstance. @embh I think you're being too literal here ;) An alternative is to put the check on token id existence in separate code before the blacklist check. That approach would fit the semantics better.

Reply:
I would definitely err on the side of keeping this literal so it's clear what the value returned from this function is telling you. Checking that the ID is in the database separately makes sense (assuming that's something we need to check).